mathesar/api/serializers/table_settings.py

```diff
@@ -39,3 +39,9 @@ def update(self, instance, validated_data):
             instance.column_order = column_order_data
             instance.save()
         return instance
+
+    def to_internal_value(self, data):
+        column_order = data.get('column_order', None)
+        if isinstance(column_order, list):
+            data['column_order'] = list(map(int, column_order))
```

Comment:
Won't setting this validation as a [ListField](https://www.django-rest-framework.org/api-guide/fields/#listfield) work?

Reply:
😲 Didn't knew setting a child would also typecast, I thought it was just type checking. Anyway, fixed now.